File: CarHomeWidget/src/com/anod/car/home/LauncherSettings.java
package com.anod.car.home;

import android.net.Uri;
import android.provider.BaseColumns;

/**
 * Settings related utilities.
 */
public class LauncherSettings {
	static final class Favorites implements BaseColumns {
        /**
         * The content:// style URL for this table
         */
		static Uri getContentUri(String packageName) {
        	if (Launcher.isFreeVersion(packageName)) {
        		return Uri.parse("content://" +
	                LauncherProvider.AUTHORITY_FREE + "/" + LauncherProvider.TABLE_FAVORITES );
        	}
    		return Uri.parse("content://" +
	                LauncherProvider.AUTHORITY_PRO + "/" + LauncherProvider.TABLE_FAVORITES );
		}
        /**
         * The content:// style URL for a given row, identified by its id.
         *
         * @param id The row id.
         *
         * @return The unique content URL for the specified row.
         */
        static Uri getContentUri(String packageName,long id) {
        	if (Launcher.isFreeVersion(packageName)) {
                return Uri.parse("content://" + LauncherProvider.AUTHORITY_FREE +
                        "/" + LauncherProvider.TABLE_FAVORITES + "/" + id
                );        		
        	}
            return Uri.parse("content://" + LauncherProvider.AUTHORITY_PRO +
                    "/" + LauncherProvider.TABLE_FAVORITES + "/" + id
            );
        }
        
        /**
         * Descriptive name of the gesture that can be displayed to the user.
         * <P>Type: TEXT</P>
         */
        static final String TITLE = "title";
        /**
         * The Intent URL of the gesture, describing what it points to. This
         * value is given to {@link android.content.Intent#parseUri(String, int)} to create
         * an Intent that can be launched.
         * <P>Type: TEXT</P>
         */
        static final String INTENT = "intent";
        /**
         * The type of the gesture
         *
         * <P>Type: INTEGER</P>
         */
        static final String ITEM_TYPE = "itemType";
        /**
         * The gesture is an application
         */
        static final int ITEM_TYPE_APPLICATION = 0;
        /**
         * The gesture is an application created shortcut
         */
        static final int ITEM_TYPE_SHORTCUT = 1;
        /**
         * The icon type.
         * <P>Type: INTEGER</P>
         */
        static final String ICON_TYPE = "iconType";
        /**
         * The icon is a resource identified by a package name and an integer id.
         */
        static final int ICON_TYPE_RESOURCE = 0;
        /**
         * The icon is a bitmap.
         */
        static final int ICON_TYPE_BITMAP = 1;
        /**
         * The icon package name, if icon type is ICON_TYPE_RESOURCE.
         * <P>Type: TEXT</P>
         */
        static final String ICON_PACKAGE = "iconPackage";
        /**
         * The icon resource id, if icon type is ICON_TYPE_RESOURCE.
         * <P>Type: TEXT</P>
         */
        static final String ICON_RESOURCE = "iconResource";
        /**
         * The custom icon bitmap, if icon type is ICON_TYPE_BITMAP.
         * <P>Type: BLOB</P>
         */
        static final String ICON = "icon";
    }
}

File: content/src/main/kotlin/info/anodsplace/carwidget/content/db/LauncherSettings.kt
package info.anodsplace.carwidget.content.db

import android.net.Uri
import android.provider.BaseColumns
import androidx.core.net.toUri


/**
 * Settings related utilities.
 */
class LauncherSettings {

    class Favorites : BaseColumns {
        companion object {
            const val TITLE = "title"
            const val INTENT = "intent"
            const val ITEM_TYPE = "itemType"
            const val ICON_TYPE = "iconType"
            const val ICON_PACKAGE = "iconPackage"
            const val ICON_RESOURCE = "iconResource"
            const val ICON = "icon"
            const val IS_CUSTOM_ICON = "isCustomIcon"

            const val ITEM_TYPE_APPLICATION = 0
            const val ITEM_TYPE_SHORTCUT = 1

            const val ICON_TYPE_RESOURCE = 0
            const val ICON_TYPE_BITMAP = 1

            fun authority(isDebug: Boolean) = if (isDebug)
                "com.anod.car.home.free.debug.shortcutsprovider"
            else
                "com.anod.car.home.free.shortcutsprovider"

            fun getContentUri(isDebug: Boolean, id: Long): Uri {
                return "$CONTENT_PREFIX${authority(isDebug = isDebug)}/$TABLE_FAVORITES/$id".toUri()
            }

            private const val CONTENT_PREFIX = "content://"
            private const val TABLE_FAVORITES = "favorites"
        }
    }
}
